packages/@react-spectrum/table/stories/CRUDExample.tsx

```diff
@@ -95,7 +95,11 @@ export function CRUDExample() {
             title="Delete"
             variant="destructive"
             primaryActionLabel="Delete"
-            onPrimaryAction={() => list.remove(dialog.item.id)}>
+            onPrimaryAction={() => {
+              // Add delay so that dialog closes and restores focus
+              // to the TableView before removing the row.
+              setTimeout(() => list.remove(dialog.item.id), 60);
```

Comment:
I don't think we're going to want this timeout
Where does focus go without the setTimeout? I think it goes to the first focusable in the scope? or does table move it to the first row?
If table does it, seems like we could consult the cached information we have an move it somewhere better
If the focus scope does it, then maybe the table can become the first focusable thing and when it gets focus, it then forwards it to the appropriate row?

Reply:
Without the timeout, focus moves to the first focusable in scope, the Select All checkbox.

Reply:
selectionManager.setFocusedKey executes within the useGrid and targets the appropriate key, but useRestoreFocus executes after, when the dialog closes.

Reply:
I think I figured this out. The problem is that the FocusScope for Dialogs and Modals always `contain` focus using `useOverlayFocusContain`, so that when the user presses the Delete button to execute the primary action that removes the item, while the `useEffect` within `useGrid` does reset focus to the appropriate `focusedKey`, the FocusScope within the Overlay always steals focus back, even if the Overlay is in the process of closing. I think we need to account for the state of the Overlay when calling useOverlayFocusContain, so that an Overlay that is closing doesn't contain focus.

Reply:
I think we were doing that at one point, when transitioning out. But I can't find it, maybe it was in some work we did but never merged.

What if, when focus is outside of a table or other collection, the children all became non-tabbable and the collection itself always received focus, then forwarded it wherever it was supposed to go?

Reply:
@snowystinger Is it this one? https://github.com/adobe/react-spectrum/pull/3704

Reply:
I can confirm that #3704 resolves the issue.

Reply:
No, it was elsewhere, the idea was that when a Popover began transitioning out, then contains would be set to false. I think it was this https://github.com/adobe/react-spectrum/pull/3725

Reply:
> No, it was elsewhere, the idea was that when a Popover began transitioning out, then contains would be set to false. I think it was this #3725

I tried incorporating the small changes in #3725, and it did not solve the problem.